type-definitions/most.js.flow

```diff
@@ -0,0 +1,410 @@
+  schedule(delay: number, period: number, task: Task): ScheduledTask;
+  cancel(task: Task): void;
+  cancelAll(predicate: (task: Task) => boolean): void;
+}
+
+export interface Disposable<A> {
+  dispose(): void | Promise<A>;
+}
+
+export interface Source<A> {
+  run (sink: Sink<A>, scheduler: Scheduler): Disposable<A>;
+}
+
+export interface Observable<A> {
+  subscribe(subscriber: Subscriber<A>): Subscription<A>;
+}
+
+export interface Subscriber<A> {
+  next(value: A): void;
+  error(err: Error): void;
+  // complete value parameter is deprecated
+  complete(value?: A): void;
+}
+
+export interface Subscription<A> {
+  unsubscribe(): void;
+}
+
+declare export class Stream<A> {
+  constructor(source: Source<A>): Stream<A>;
+  source: Source<A>;
+  run (sink: Sink<A>, scheduler: Scheduler): Disposable<A>;
+  reduce<B>(f: (b: B, a: A) => B, b: B): Promise<B>;
+  observe(f: (a: A) => any): Promise<any>;
+  forEach(f: (a: A) => any): Promise<any>;
+  drain(): Promise<any>;
+  subscribe(subscriber: Subscriber<A>): Subscription<A>;
+
+  constant<B>(b: B): Stream<B>;
+  map<B>(f: (a: A) => B): Stream<B>;
+  tap(f: (a: A) => any): Stream<A>;
+  chain<B>(f: (a: A) => Stream<B>): Stream<B>;
+  flatMap<B>(f: (a: A) => Stream<B>): Stream<B>;
+
+  ap<B, C>(fs: Stream<(a: A) => B>): Stream<C>;
+
+  // Note: Without higher-kinded types, the types for these
+  // cannot be written in a completely safe manner; See https://github.com/Microsoft/TypeScript/issues/1290
+  // For better type safety, consider using most.join/switch/switchLatest with thru
+  join(): A;
+  switch(): A;
+  switchLatest(): A;
+
+  continueWith(f: (a: any) => Stream<A>): Stream<A>;
+  concatMap<B>(f: (a: A) => Stream<B>): Stream<B>;
+  mergeConcurrently<B>(concurrency: number): Stream<B>;
+  merge(...ss: Array<Stream<A>>): Stream<A>;
+  mergeArray(streams: Array<Stream<A>>): Stream<A>;
+
+  combine<B, R>(
+    fn: (a: A, b: B) => R,
+    b: Stream<B>
+  ): Stream<R>;
+  combine<B, C, R>(
+    fn: (a: A, b: B, c: C) => R,
+    b: Stream<B>,
+    c: Stream<C>
+  ): Stream<R>;
+  combine<B, C, D, R>(
+    fn: (a: A, b: B, c: C, d: D) => R,
+    b: Stream<B>,
+    c: Stream<C>,
+    d: Stream<D>
+  ): Stream<R>;
+  combine<B, C, D, E, R>(
+    fn: (a: A, b: B, c: C, d: D, e: E) => R,
+    b: Stream<B>,
+    c: Stream<C>,
+    d: Stream<D>,
+    e: Stream<E>
+  ): Stream<R>;
+
+  combineArray<B, R>(
+    fn: (a: A, b: B) => R,
+    streams: [Stream<B>]
+  ): Stream<R>;
+  combineArray<B, C, R>(
+    fn: (a: A, b: B, c: C) => R,
+    streams: [Stream<B>, Stream<C>]
+  ): Stream<R>;
+  combineArray<B, C, D, R>(
+    fn: (a: A, b: B, c: C, d: D) => R,
+    streams: [Stream<B>, Stream<C>, Stream<D>]
+  ): Stream<R>;
+  combineArray<B, C, D, E, R>(
+    fn: (a: A, b: B, c: C, d: D, e: E) => R,
+    streams: [Stream<B>, Stream<C>, Stream<D>, Stream<E>]
+  ): Stream<R>;
+  combineArray<V, R>(
+    fn: (a: A, ...rest: V[]) => R,
+    streams: Stream<V>[]
+  ): Stream<R>;
```

Comment:
The problem is that typescript requires you to specify more types, but flow mostly can infer them and if it does not ask to specify type (it's not exported) then everything should work correctly.